client/state/themes/selectors/does-theme-require-atomic-site.ts

```diff
@@ -0,0 +1,13 @@
+import { isExternallyManagedTheme } from './is-externally-managed-theme';
+import 'calypso/state/themes/init';
+
+/**
+ * Checks if a theme requires an Atomic site.
+ *
+ * @param state The app global state
+ * @param themeId The theme ID
+ * @returns True if the theme requires an Atomic site
+ */
+export function doesThemeRequireAtomicSite( state = {}, themeId: string ): boolean {
```

Comment:
Multiple comments here:
1. Can we use the `AppState` type for `state`?
2. Can we add types to the JSDoc?